File: model/src/main/java/io/github/runtimemodels/chazm/event/PossessesEventFactory.java
package io.github.runtimemodels.chazm.event;

import io.github.runtimemodels.chazm.relation.Possesses;

/**
 * The {@linkplain PossessesEventFactory} interface defines the API for constructing {@linkplain PossessesEvent}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface PossessesEventFactory {

	/**
	 * Constructs an {@linkplain PossessesEvent}.
	 *
	 * @param category
	 *            the {@linkplain EventCategory}.
	 * @param possesses
	 *            the {@linkplain Possesses}.
	 *
	 * @return a {@linkplain PossessesEvent}.
	 */
	PossessesEvent build(EventCategory category, Possesses possesses);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/relation/PossessesEventFactory.kt
package runtimemodels.chazm.model.relation

import runtimemodels.chazm.api.relation.Possesses
import runtimemodels.chazm.model.event.EventType

/**
 * The [PossessesEventFactory] interface defines the API for constructing [PossessesEvent]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface PossessesEventFactory {
    /**
     * Constructs an [PossessesEvent].
     *
     * @param category  the [EventType].
     * @param possesses the [Possesses].
     * @return a [PossessesEvent].
     */
    fun build(category: EventType, possesses: Possesses): PossessesEvent
}
